render-only-sample/rosumd/RosUmdResource.cpp

```diff
@@ -263,6 +263,116 @@ RosUmdResource::Unmap(
 }
 
 void
+RosUmdResource::FillTileInfo(UINT bpp)
+{
+    // Provide detailed information about tile.
+    // Partial information about 4kB tiles, 1kB sub-tiles and micro-tiles for
+    // given bpp is precalculated.
+    // Values are used i.e. during converting bitmap to tiled texture
+
+    if (bpp == 8)
+    {
+        m_TileInfo.VC4_1kBSubTileWidthPixels        = VC4_1KB_SUB_TILE_WIDTH_8BPP;
+        m_TileInfo.VC4_1kBSubTileHeightPixels       = VC4_1KB_SUB_TILE_HEIGHT_8BPP;
+        m_TileInfo.VC4_MicroTileWidthBytes          = VC4_MICRO_TILE_WIDTH_BYTES_8BPP;
+        m_TileInfo.vC4_MicroTileHeight              = VC4_MICRO_TILE_HEIGHT_8BPP;
+    }
+    else
+    if (bpp == 16)
+    {
+        m_TileInfo.VC4_1kBSubTileWidthPixels        = VC4_1KB_SUB_TILE_WIDTH_16BPP;
+        m_TileInfo.VC4_1kBSubTileHeightPixels       = VC4_1KB_SUB_TILE_HEIGHT_16BPP;
+        m_TileInfo.VC4_MicroTileWidthBytes          = VC4_MICRO_TILE_WIDTH_BYTES_16BPP;
+        m_TileInfo.vC4_MicroTileHeight              = VC4_MICRO_TILE_HEIGHT_16BPP;
+    }
+    else
+    if (bpp == 32)
+    {
+        m_TileInfo.VC4_1kBSubTileWidthPixels        = VC4_1KB_SUB_TILE_WIDTH_32BPP;
+        m_TileInfo.VC4_1kBSubTileHeightPixels       = VC4_1KB_SUB_TILE_HEIGHT_32BPP;
+        m_TileInfo.VC4_MicroTileWidthBytes          = VC4_MICRO_TILE_WIDTH_BYTES_32BPP;
+        m_TileInfo.vC4_MicroTileHeight              = VC4_MICRO_TILE_HEIGHT_32BPP;
+    }
+    
+    // Calculate sub-tile width in bytes
+    m_TileInfo.VC4_1kBSubTileWidthBytes = m_TileInfo.VC4_1kBSubTileWidthPixels * (bpp / 8);
+    
+    // 4kB tile consists of four 1kB sub-tiles
+    m_TileInfo.VC4_4kBTileWidthPixels = m_TileInfo.VC4_1kBSubTileWidthPixels * 2;
+    m_TileInfo.VC4_4kBTileHeightPixels = m_TileInfo.VC4_1kBSubTileHeightPixels * 2;
+    m_TileInfo.VC4_4kBTileWidthBytes = m_TileInfo.VC4_1kBSubTileWidthBytes * 2;
+}
+
+void MapDxgiFormatToInternalFormats(DXGI_FORMAT format, UINT &bpp, RosHwFormat &rosFormat)
+{
+
+    // Number of HW formats is limited, so some of DXGI formats must be emulated.
+    // For example, format DXGI_FORMAT_R8_UNORM is emulated with  DXGI_FORMAT_R8G8B8A8_UNORM 
+    // where G8, B8 are set to 0
+    // 
+    switch (format)
+    {
+    case DXGI_FORMAT_R8G8B8A8_UNORM:
+    {
+        bpp = 32;
+        rosFormat = RosHwFormat::X8888;
+    }
+    break;
+
+    case DXGI_FORMAT_R8G8_UNORM:
+    {
+        bpp = 32;
+        rosFormat = RosHwFormat::X8888;
+    }
+    break;
+
+    case DXGI_FORMAT_R8_UNORM:
+    {
+        bpp = 32;
+        rosFormat = RosHwFormat::X8888;
+    }
+    break;
+
+    case DXGI_FORMAT_A8_UNORM:
+    {
+        bpp = 32;
+        rosFormat = RosHwFormat::X8888;
+    }
+    break;
+
+    default:
+        bpp = 8;
+        rosFormat = RosHwFormat::X8;
+    }
+}
+
+void
+RosUmdResource::CalculateTilesInfo()
+{
+    UINT bpp = 0;
```

Comment:
Variables that are immediately passed as an _Out_ parameter can be declared without initializing them, since _Out_ requires that the function write to the parameter before returning.